### ISSUE TITLE
Get Bitfinex error

### ISSUE THREAD

**Issue by StyleShit** [bitfinexcom/bitfinex-api-node#169]
When I get errors frp, BItfinex, such as "order not found" or something,
How do I get them in my code? I'm using WebSockets V2

Thnx

**Comment by f3rno** [comment]
@StyleShit it depends on the error, but you can listen for 'error' events with `ws2.on('error', (err) => ...`

If it's an error that would appear as a notification on the platform, it will come in as a notification on `ws2` with the status set to `ERROR`. For example, cancelling an unknown order:

```js
[
  0,
  "n",
  [
    1514817408519,
    "oc-req",
    null,
    null,
    [
      0,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      0,
      null,
      null
    ],
    null,
    "ERROR",
    "id: invalid"
  ]
]
```

**Comment by StyleShit** [comment]
@f3rno, the on("error") doesn't work. I mean the notifications.
If I understand you right, I should catch on("message") and check if there is "ERROR" in the response array?

**Comment by f3rno** [comment]
@StyleShit aye, you need to specifically check for `'n'` messages on the auth channel (0) with a `STATUS` field of '"ERROR"`. The different fields are documented here: https://docs.bitfinex.com/v2/reference#ws-auth-notifications

**Comment by StyleShit** [comment]
@f3rno Thank you for the detailed answer!
I'll check it out.

**Comment by StyleShit** [comment]
@f3rno OK, I've got the errors, but it was too complicated.. 
( checking that it's an array, channel 0, 'n', status ERROR, etc. ).
Can you add `on('n')` listener which will return a JSON with the status and message to the WebSocket?

Anyway, thank you very much for your help! You are awesome :)

**Comment by f3rno** [comment]
@StyleShit no problem! I'm happy to help, hehe.

We've added notification listeners in `v2.0.0-beta` [here](https://github.com/f3rno/bitfinex-api-node/blob/v2-refactor/lib/transports/ws2.js#L1716) which will be released very soon (today/tomorrow) :)

If you use the v2 beta and want to check for order errors, you can catch them on the returned promise:

```js
ws.submitOrder(order).then(() => {
  // success
}).catch((err) => {
  // invalid ID, insufficient balance, etc
})
```

Checking for new order notifications is done with:

```js
ws.onNotification({ type: 'on-req' }, (notification) => {
  // check status field, etc
})
```

**Comment by StyleShit** [comment]
@f3rno
When you say "released", you mean that it will be released as a stable version in the master branch? 
(sorry if I'm saying that wrong.. I'm pretty new to github :P ) 
Coz I prefer a stable version instead of beta. I'm working on a bot which should be as reliable as I can.

**Comment by f3rno** [comment]
@StyleShit released as in on master and tagged as `v2.0.0-beta`; the `beta` suffix is because there are many new changes, some of which are a bit opinionated, and we want to get some user feedback/testing in before finally pushing `2.0.0` and freezing the API.

It is pretty stable, and has features which make bot development a bit easier, so you might want to try it :) You can find it here for now: https://github.com/f3rno/bitfinex-api-node/tree/v2-refactor

There are snippets in the `examples/` folder, and API docs in README.md and `doc/` so you can see what's different.

**Comment by f3rno** [comment]
@StyleShit `v2.0.0-beta` has been released, so I'll close this issue. If you still need help just continue the thread here.
